charts/internal/cilium/values.yaml

```diff
@@ -0,0 +1,373 @@
+requirements:
+  # Include the cilium-agent DaemonSet
+  agent:
+    enabled: true
+    # Do not run Cilium agent when running with clean mode. Useful to completely
+    # uninstall Cilium as it will stop Cilium from starting and create artifacts
+    # in the node.
+    sleepAfterInit: false
+    # Keep the deprecated selector labels when deploying Cilium DaemonSet
+    keepDeprecatedLabels: false
+
+  # Include the cilium-config ConfigMap
+  config:
+    enabled: true
+
+  # Include the cilium-operator Deployment
+  operator:
+    enabled: true
+
+  # Include the PreFlight DaemonSet
+  preflight:
+    enabled: false
+    # Path to write the --tofqdns-pre-cache file to. When empty no file is
+    # generated.
+    tofqdnsPreCache: ""
+
+  # Include the Hubble deployment
+  hubble:
+    enabled: false
+
+# global groups all configuration options that have effect on all sub-charts
+global:
+  # pullPolicy is the container image pull policy
+  pullPolicy: IfNotPresent
+
+  # etcd is the etcd configuration
+  etcd:
+    # enabled enables use of etcd
+    enabled: true
+
+    # managed turns on managed etcd mode based on the cilium-etcd-operator
+    managed: true
```

Comment:
sure but then you need to another loadbalancer which is not desirable, also the values you see here are the defaults, we override those with new defaults  which sets both to false. 

I have already tested the cilium-etcd-operator and it does not work well with hibernation in our case, so we are now using CRDs (i.e., our etcd) for storage, and during tested that was much better. 

As a final step, I actually don't mind using druid, but for a first version, I think this should suffice, WDYT?

Reply:
> The values you see here are the defaults, we override those with new defaults which sets both to false.

Yes, i noticed that we set default to `false`. And i'm fine with it, for getting started with cilium. But ⚠️ in production while deploying i expect we are planing to enable etcd 
for reason mentioned [here](https://cilium.readthedocs.io/en/stable/gettingstarted/k8s-install-external-etcd/#when-do-i-need-to-use-a-kvstore).

> I have already tested the cilium-etcd-operator and it does not work well with hibernation in our case, so we are now using CRDs (i.e., our etcd) for storage, and during testing that was much better.

Possible. Cause, CRDs got additional caching mechanism from kube-apiserver over etcd. But again not sure for large cluster. Also, if we are going for CRD, then i'll prefer having separate etcd configured on [kube-apisrver](https://github.com/gardener/gardener/blob/master/charts/seed-controlplane/charts/kube-apiserver/templates/kube-apiserver.yaml#L99) with `cilium` prefix to reduce load from etcd-main. But to counter this, it will loose the abstraction provided by network extensibility. (unless we introduce some webhook in this extension which might looks little ugly solution to think off).

> As a final step, I actually don't mind using druid, but for a first version, I think this should suffice, WDYT?
I'm not sure. I don't have good feeling overall, if we are not going for etcd based on docs. And as you mentioned, there is issue with hibernation when etcd is enabled. 
But for first cut, to evaluate with real time workload, i'm always fine with taking progressive approach.

Reply:
okay, then I will raise an issue on this Repo to have druid be our `cilium-etcd-operator`, and for now we can start with with the CRD approach as it is the safest, WDYT?

Reply:
Does this require any change in the cilium side?

Reply:
Not really, Cilium support the idea of an external un-managed ETCD, in this case, it will be an operator managed by Gardener.

Reply:
Ok thanks for the information.